### PR TITLE
[WIP] AUTH-89: Audit logs as opts, yet another PR

### PR DESCRIPTION
**Pull request by ibihim**
## What

Start `oauth-server` with audit options.

## Why

Enable audit logging for `oauth-server` and authentication events.

### CODE REVIEW

**Comment by ibihim** [comment]
@stlaz, do you have a clue, why the `audit-policy.yaml` doesn't create a `configmap`?

**Comment by EmilyM1** [comment]
It may be possible that CAO is no longer creating static policies, but only creating dynamic policies

File: bindata/oauth-openshift/deployment.yaml [hunk 1]
```diff
@@ -63,15 +63,25 @@ spec:
...
           securityContext:
+            privileged: true
             readOnlyRootFilesystem: false # because of the `cp` in args
             runAsUser: 0 # because /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem is only writable by root
           volumeMounts:
+            - mountPath: /var/run/configmaps/audit
+              name: audit-policies
+            - mountPath: /var/log/oauth-server
+              name: audit-dir
             - name: v4-0-config-system-session
```
Comment (EmilyM1) [inline bindata/oauth-openshift/deployment.yaml:85]: These are both names similar but are different audit logs:
https://github.com/openshift/cluster-authentication-operator/blob/master/bindata/oauth-apiserver/deploy.yaml#L101
Wonder if unique name would be best?

Comment (ibihim) [reply bindata/oauth-openshift/deployment.yaml:85]: This is done on purpose as we are writing audit logs to the node or am I wrong here?

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -260,6 +260,7 @@ func prepareOauthOperator(controllerContext *controllercmd.ControllerContext, op
 		"OpenshiftAuthenticationStaticResources",
 		assets.Asset,
 		[]string{
+			"oauth-openshift/audit-policy.yaml",
```
Comment (ibihim) [inline pkg/operator/starter.go:263]: I am not sure if I am missing something here, but it seems like the configmap is not there, when I push the CAO image into the cluster. Am I missing something?

@emilym1 mentioned that there are no static configmaps anymore?

**Pull request closed without merging**